### PR TITLE
Rename the n1k module directory

### PR DESCRIPTION
**Pull request by imcsk8**
Change the name of the n1k module directory from n1k-vsm to n1k_vsm
as asked by this bug:
https://bugzilla.redhat.com/show_bug.cgi?id=1195869

The name of this module in openstack-puppet rpm is n1k-vsm. Seems like
puppet doesnt like that module name. Could you please change it to
n1k_vsm(not "-" its "_" )?

This puppet module is only used by cisco with "Generic RHEL 7" host
group and does not affect any other puppet classes or installer logic.

### CODE REVIEW

**Comment by strider** [comment]
Travis is failing because you need to rename the directory as well!

**Comment by imcsk8** [comment]
this is odd, i did rename the directory. you can see it in the files changed section

**Comment by imcsk8** [comment]
actually the test is failing because the directory name change, how can we fix this?

Missing ./n1k-vsm/Rakefile, not running spec tests. You have to manually check whether this is OK.

**Comment by javierpena** [comment]
I think this is an issue in the .travis.sh script. It tries to check which modules should be tested by doing a "git diff HEAD~1 Puppetfile", which in this case shows the old and the new names. n1k_vsm tests work fine, but there is no n1k_vsm directory, so it fails.

tl;dr: I think we can just merge this one if the cisco testing is fine for them and does not break Packstack or Staypuft.

**Comment by javierpena** [comment]
Tested with Packstack. As expected nothing was broken.

**Comment by paramite** [comment]
Agree on merging. I don't see any problem with this PR.
